libs/partners/huggingface/langchain_huggingface/llms/huggingface_pipeline.py

```diff
@@ -125,10 +126,41 @@ def from_model_id(
                         model = OVModelForCausalLM.from_pretrained(
                             model_id, export=True, **_model_kwargs
                         )
+                elif backend == "ipex":
+                    try:
+                        import torch
+                        from optimum.intel.ipex import (  # type: ignore[import]
+                            IPEXModelForCausalLM,
+                        )
+                    except ImportError:
+                        raise ValueError(
+                            "Could not import optimum-intel python package. "
+                            "Please install it with: "
+                            "pip install 'optimum[ipex]' "
+                            "or follow installation instructions from: "
+                            " https://github.com/rbrugaro/optimum-intel "
+                        )
+                    try:
+                        # use TorchScript model
+                        config = AutoConfig.from_pretrained(model_id)
+                        export = not getattr(config, "torchscript", False)
+                    except RuntimeError:
+                        logger.warning(
+                            "We will use IPEXModel with export=True to export the model"
+                        )
+                        export = True
+                    model = IPEXModelForCausalLM.from_pretrained(
+                        model_id,
+                        export=export,
```

Comment:
The IPEXModel only support export=True for now, it means we will load the model in HF and do ipex jit trance.

Reply:
@jiqing-feng is your recommendation to remove the export setting logic or leave it even if for now it will render export=True? I followed same logic as in[ /optimum/intel/pipelines/pipeline_base.py](https://github.com/huggingface/optimum-intel/blob/2c79d987b685ef5e87b2062518598e9c1cf64c78/optimum/intel/pipelines/pipeline_base.py#L117 )

Reply:
I see, yes you can detect export in this way.